test/functional/keyed_items.js

```diff
@@ -169,6 +169,23 @@ describe('rendering with keys', () => {
     expect(firstNode.parentNode).to.equal(null);
   });
 
+  it('should not throw error if rendering content on parent with outdated key map', () => {
+    function render(tag) {
+      elementVoid(tag, 'key');
+    }
+
+    patch(container, render, 'div');
+    const firstNode = container.childNodes[0];
+
+    container.removeChild(container.childNodes[0]);
+
+    expect(() => patch(container, render, 'span')).to.not.throw(Error);
+    const newNode = container.childNodes[0];
+    expect(newNode).not.to.equal(firstNode);
```

Comment:
That wouldn't cause the test to fail without my change though, the use case that it's fixing involves patching with content that uses the same key as a previous patch, but doesn't match the child stored in the parent's key map. In that case incremental dom will try to remove that previous keyed child from the parent, triggering an error because the child is not already there. So this is what this is the use case this test is reproducing.

What we need to check is just that the error is not thrown in this case, though it might be useful to also make sure that the content is rendered correctly too, which is why I added these other assertions as well.